frontend/packages/console-shared/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -140,52 +143,76 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
     return true;
   }
 
-  private getDropdownList = (
-    {
+  private craftResourceKey = (
+    resource: K8sResourceKind,
+    props: ResourceDropdownProps,
+  ): { customKey: any; key: any } => {
+    const { customResourceKey, resourceFilter, dataSelector } = props;
+    let key;
+    if (resourceFilter && resourceFilter(resource)) {
+      key = _.get(resource, dataSelector);
+    } else if (!resourceFilter) {
+      key = _.get(resource, dataSelector);
```

Comment:
use optional chaining here

Reply:
dataSelector here is a prop which is been passed to this Resource dropdown component. The values of the dataSelector will be an array. 
eg: ```dataSelector={['metadata', 'labels', 'app.kubernetes.io/part-of']}```.  Hence the lodash _.get.